DNN Platform/Library/Application/DotNetNukeContext.cs

```diff
@@ -1,41 +1,40 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information
+
 namespace DotNetNuke.Application
-{
-    using System.Collections.Generic;
+{
+    using System;
+    using System.Collections.Generic;
 
-    using DotNetNuke.Collections.Internal;
+    using DotNetNuke.Abstractions.Application;
+    using DotNetNuke.Collections.Internal;
+    using DotNetNuke.Common;
     using DotNetNuke.UI.Containers.EventListeners;
     using DotNetNuke.UI.Skins.EventListeners;
 
+    using Microsoft.Extensions.DependencyInjection;
+
     /// <summary>
     /// Defines the context for the environment of the DotNetNuke application.
     /// </summary>
-    public class DotNetNukeContext
+    public class DotNetNukeContext : IDnnContext
     {
-        private static DotNetNukeContext _current;
-        private readonly Application _application;
-        private readonly IList<ContainerEventListener> _containerEventListeners;
-        private readonly IList<SkinEventListener> _skinEventListeners;
-
-        /// <summary>
-        /// Initializes a new instance of the <see cref="DotNetNukeContext" /> class.
-        /// </summary>
-        protected DotNetNukeContext()
```

Comment:
@mitchelsellers do you have any concern about a breaking change from removing this `protected` constructor?

Reply:
@bdukes is this a breaking change? the class uses a `public` access modifier, by removing the `protected` constructor it opens the API structure up slightly. I had to remove the `protected` constructor so the Dependency Injection container could properly resolve `DotNetNukeContext`. If we want to keep this `protected` I could create a new child class with a public constructor and then register `IDnnContext` to the child class.

Reply:
The scenario that would break is when someone is inheriting from `DotNetNukeContext` and calling `base()` in its constructor.  It shouldn't be breaking to leave these two `protected` constructors and add the new `public` constructor, though.

Reply:
I just tried adding 2 default constructors
```
public DotNetNukeContext() {}
protected DotNetNukeContext() {}
```

This has a compilation error as a class can only have 1 parameterless default constructor regardless of access level. I don't think this is a breaking change because child classes will automatically call the default constructor regardless if it is public or protected. If the constructor was private, then you start getting into different behaviors.

Reply:
I'm not seeing a default constructor on the new version of `DotNetNukeContext`, just `public DotNetNukeContext(IApplicationInfo applicationInfo)`, am I missing it?

Reply:
🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️

Glad we talked through this, my mistake. You are 100% correct, there is no long a default constructor and that is **wrong**. We need to add a public default constructor to prevent a breaking change.

Reply:
I have added the protected constructor back in, but have deprecated it. Marking this conversation as resolved.